StormyHares/BangazonCLI
# Stale products cm

* **The Name of this branch is**
staleProducts-cm

* **Please check if the PR fulfills these requirements**



* **What kind of change does this PR introduce?** (Bug Fix, Feature, Document Update...)
FEATURE - Stale Product Menu is now functional
UPDATE - ReadMe to reflect feature added and project installation instructions


* **What Ticket is associated with this PR**
#9 


* **What is the new behavior?**  (if this is a feature change)
When option 4 is selected from the main menu the user is presented with a report of all products that meet any of the criteria outlined in ticket #9

* **List all core functionality** (Must include at least one Unit Test for Each)
No new core functionality

* **Other information / Steps to Test**:
1. git fetch
1. git checkout staleProducts-cm
1. navigate to test directory and confirm all tests are still passing
1. navigate to src code directory and initialize project with dotnet run
1. select option 4 from the main menu - you should receive a message stating that there are no stale products to display
1. To simulate requirement 1 from #10 open your database browser and execute the following sql script
```
INSERT INTO Product
VALUES(null, 1, 'STALE REQ 1', 'Not ordered older than 180 days', 42134,  50, '2017-02-12');
```
7. Save the database file and in the terminal return to the main menu and reselect the stale product menu (option 4)
1. To simulate requirement 2 from #10 open your database browser and execute the following sql script
```
INSERT INTO Product
VALUES(null, 2, "STALE REQ 2", "Added to an order but not purchased and the order is more than 90 days old", 42134,  33, '2017-10-02');

INSERT INTO Orders
VALUES(null, 1, null, '2017-10-08', null);

INSERT INTO OrderProduct
SELECT null, o.Id, p.Id
FROM Orders o, Product p
WHERE o.Created = '2017-10-08'
AND p.Title = 'STALE REQ 2';
```
9. Save the database file and in the terminal return to the main menu and reselect the stale product menu (option 4)
1. To simulate requirement 3 from #10 open your database browser and execute the following sql script
```

INSERT INTO Product
VALUES(null, 3, "STALE REQ 3", "Purchased, but with remaining quantity and it is more than 180 days old", 42134,  2, '2017-01-12');

INSERT INTO Orders
VALUES(null, 1, 1, '2017-12-12', '2017-12-12');

INSERT INTO OrderProduct
SELECT null, o.Id, p.Id
FROM Orders o, Product p
WHERE o.Created = '2017-12-12'
AND p.Title = 'STALE REQ 3';
```
